runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecord.java

```diff
@@ -189,4 +204,11 @@ void serialize(ByteBuf b) {
         serializedSize = b.readableBytes() + 4;
         b.writeInt(serializedSize);
     }
+
+    @Data
+    @AllArgsConstructor
+    static public class SMRRecordInfo {
+        SMRRecordLocator locator;
+        int serializedSize;
```

Comment:
Why `serializedSize` is not part SMRRecordLocator? What's the reason behind to have a separate container(SMRRecordInfo)? You can directly set `serializedSize` into `SMRRecordLocator`  later right?

Reply:
I need a container to maintain size information in the object such that the object could set the serializedSize into SMRGarbageRecord. I think it is not a good idea to consolidate size information into location information.